### ISSUE TITLE
Unable to build from Skypack or ESM.sh non-versioned import URL's

### ISSUE THREAD

**Issue by brickpop** [denoland/dnt#179]
Trying to build a very basic example, using SkyPack/ESM.sh:

```ts
import { Wallet } from "https://cdn.skypack.dev/@ethersproject/wallet";

console.log(Wallet.createRandom().privateKey);
```

This code will work fine with `deno run`, however trying to build it with the [suggested settings](https://github.com/denoland/dnt#setup) will throw an error:

```
[dnt] Transforming...
[dnt] Running npm install...
npm ERR! code EINVALIDPACKAGENAME
npm ERR! Invalid package name "-/" of package "-/@ethersproject": name can only contain URL-friendly characters.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/user/.npm/_logs/2022-06-17T23_38_09_035Z-debug-0.log
error: Uncaught (in promise) Error: npm install failed with exit code 1
        throw new Error(
              ^
    at runCommand (https://deno.land/x/dnt@0.26.0/lib/utils.ts:56:15)
    at async build (https://deno.land/x/dnt@0.26.0/mod.ts:183:5)
    at async file:///home/user/dev/tests/dnt/build-npm.ts:5:1
```

I have tried using an esm.sh URL, but a similar error appears:

```ts
import { Wallet } from "https://esm.sh/@ethersproject/wallet";

console.log(Wallet.createRandom().privateKey);
```

Build output:

```
[dnt] Transforming...
[dnt] Running npm install...
npm ERR! code EINVALIDPACKAGENAME
npm ERR! Invalid package name "v86/" of package "v86/@ethersproject": name can only contain URL-friendly characters.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/user/.npm/_logs/2022-06-17T23_40_38_188Z-debug-0.log
error: Uncaught (in promise) Error: npm install failed with exit code 1
        throw new Error(
              ^
    at runCommand (https://deno.land/x/dnt@0.26.0/lib/utils.ts:56:15)
    at async build (https://deno.land/x/dnt@0.26.0/mod.ts:183:5)
    at async file:///home/user/dev/tests/dnt/build-npm.ts:5:1
```

Getting this step to work would mean everything to be able to make the jump to Deno <3
Thank you!

**Comment by brickpop** [comment]
The problem happens on NPM, but for some reason, Skypack URL's are interpreted such in a way that NPM is asked to install something like this (package.json):

```json
{
  ...
  "dependencies": {
    "-/": "ethersproject"
  },
}
```

Both with the intended URL (`https://cdn.skypack.dev/@ethersproject/wallet`) and with the expanded one (`https://cdn.skypack.dev/-/@ethersproject/wallet@v5.6.2-hbht5UfbVmWkq5PkNraB/dist=es2019,mode=imports/optimized/@ethersproject/wallet.js`).

**Comment by dsherret** [comment]
It's hard to detect the npm version for these, so this will just default to include the source for these.

To fix this, you can (and should) use a versioned skypack or esm.sh url, such as: https://esm.sh/@ethersproject/wallet@5.6.2

**Comment by alexgleason** [comment]
I'm getting a similar error, but I'm using a URL in the "correct" format:

```
➜  strfry-policies git:(npm) deno task npm 0.1.0
Task npm deno run -A scripts/npm.ts "0.1.0"
A new release of Deno is available: 1.32.1 → 1.32.3 Run `deno upgrade` to install it.
[dnt] Transforming...
[dnt] Running npm install...
npm ERR! code EINVALIDPACKAGENAME
npm ERR! Invalid package name "v114/nostr-tools" of package "v114/nostr-tools@1.8.4": name can only contain URL-friendly characters.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/alex/.npm/_logs/2023-04-16T16_52_42_536Z-debug-0.log
error: Uncaught (in promise) Error: npm install failed with exit code 1
        throw new Error(
              ^
    at runCommand (https://deno.land/x/dnt@0.34.0/lib/utils.ts:57:15)
    at async build (https://deno.land/x/dnt@0.34.0/mod.ts:210:5)
    at async file:///home/alex/Projects/strfry-policies/scripts/npm.ts:5:1
```
